app/services/support_interface/ministerial_report_export.rb

```diff
@@ -0,0 +1,58 @@
+module SupportInterface
+  class MinisterialReportExport
+    def call
+      results = subject_status_count
+      subject_mapping(results)
+
+      subject_count = Subject::SUBJECTS.count
+
+      rows = Array.new(subject_count, nil)
+      subject_count.times do |n|
+        rows[n] = {
+          subject: Subject::SUBJECTS[n],
+          applications: 0,
+          offer_received: 0,
+          accepted: 0,
+          application_declined: 0,
+          application_rejected: 0,
+        }
+      end
+
+      stem_index = Subject::SUBJECTS.index(:stem)
+      ebacc_index = Subject::SUBJECTS.index(:ebacc)
+      secondary_index = Subject::SUBJECTS.index(:secondary)
+
+      results.each do |key, count|
+        subject, status = key
+        subject_index = Subject::SUBJECTS.index(subject)
+
+        mapped_statuses = Subject::STATUS_MAPPING[status]
+        mapped_statuses.each do |mapped_status|
+          rows[stem_index][mapped_status] += count if Subject::STEM_SUBJECTS.include? subject
+          rows[ebacc_index][mapped_status] += count if Subject::EBACC_SUBJECTS.include? subject
+          rows[secondary_index][mapped_status] += count if Subject::SECONDARY_SUBJECTS.include? subject
+          rows[subject_index][mapped_status] += count
+        end
+      end
+      rows
+    end
+
+    alias data_for_export call
+
+  private
+
+    def subject_status_count
+      Subject
+        .joins(courses: :application_choices)
+        .group('subjects.code', 'application_choices.status')
```

Comment:
Yea this is interesting, because say a candidate gets 3 offers and accepts one, then they will have 2 declined choices. Having said that, i guess that's what the 'candidates' section of the export of the export is for.